src/playercards/AllCardsBag.ttslua

```diff
@@ -147,89 +333,216 @@ function buildSupplementalIndexes()
+        local usesCount = cardMetadata.uses[1].count or -1
+        if not usesIndex[usesCount] then
+          usesIndex[usesCount] = { }
+        end
+        table.insert(usesIndex[usesCount], cardMetadata.id)
+        if not usesIndex[string.lower(cardMetadata.uses[1].type)] then
+          usesIndex[string.lower(cardMetadata.uses[1].type)] = { }
+        end
+        table.insert(usesIndex[string.lower(cardMetadata.uses[1].type)], cardMetadata.id)
       end
+      -- Old Taboo Code
+      -- if bonded then
+      --   for _, bondData in ipairs(bonded) do
+      --     local bondCardMetadata = cardIdIndex[bondData.id].metadata
+      --     if bondCardMetadata.uses then
+      --       if not usesIndex[bondCardMetadata.uses[1].count] then
+      --         usesIndex[bondCardMetadata.uses[1].count] = { }
+      --       end
+      --       table.insert(usesIndex[bondCardMetadata.uses[1].count], cardMetadata.id)
+      --       if not usesIndex[string.lower(bondCardMetadata.uses[1].type)] then
+      --         usesIndex[string.lower(bondCardMetadata.uses[1].type)] = { }
+      --       end
+      --       table.insert(usesIndex[string.lower(bondCardMetadata.uses[1].type)], cardMetadata.id)
+      --     end
+      --   end
+      -- end
+        
     end
   end
-  for _, indexTable in pairs(classAndLevelIndex) do
-    table.sort(indexTable, cardComparator)
+  
+  for _, indexTable in pairs(classIndex) do
+    sortCardsTable(indexTable)
   end
   for _, indexTable in pairs(cycleIndex) do
-    table.sort(indexTable)
+    sortCardsTable(indexTable)
+  end
+  for _, indexTable in pairs(traitsIndex) do
+    sortCardsTable(indexTable)
   end
-  table.sort(basicWeaknessList, cardComparator)
-  table.sort(uniqueWeaknessList, cardComparator)
+  for _, indexTable in pairs(specialCriteriaIndex) do
+    sortCardsTable(indexTable)
+  end
+  sortCardsTable(basicWeaknessList)
+  sortCardsTable(uniqueWeaknessList)
 end
 
 -- Comparison function used to sort the class card bag indexes.  Sorts by card
 -- level, then name, then subname.
 function cardComparator(id1, id2)
   local card1 = cardIdIndex[id1]
   local card2 = cardIdIndex[id2]
+  local card1Level = card1.metadata.level or -1
+  local card2Level = card2.metadata.level or -1
+  local card1Class = card1.metadata.class or "X"
+  if string.find(card1Class, "|") then
+    card1Class = "Multi"
+  end
+  local card2Class = card2.metadata.class or "X"
+  if string.find(card2Class, "|") then
+    card2Class = "Multi"
+  end
+  local cardClassOrder = {
+    X = 0,
+    Guardian = 1,
+    Seeker = 2,
+    Rogue = 3,
+    Mystic = 4,
+    Survivor = 5,
+    Neutral = 6,
+    Multi = 7
+  }
+  card1Class = cardClassOrder[card1Class]
+  card2Class = cardClassOrder[card2Class]
 
-  if (card1.metadata.level ~= card2.metadata.level) then
-    return card1.metadata.level < card2.metadata.level
+  if (card1Level ~= card2Level) then
+    return card1Level < card2Level
+  end
+  if card1Class ~= card2Class then
```

Comment:
Could use `elseif` to make this shorter

Reply:
Making the change, but how does it make it shorter?  Isn't it the same number of lines & same execution time, just semantically different?

Reply:
You save an `end` statement for each `elseif`:

```lua
if card1Level ~= card2Level then
    return card1Level < card2Level
elseif card1Class ~= card2Class then
    return card1Class < card2Class
elseif card1.data.Nickname ~= card2.data.Nickname then
    return card1.data.Nickname < card2.data.Nickname
else
   return card1.data.Description < card2.data.Description
end
```